src/auth/auth.service.ts

```diff
@@ -102,6 +128,50 @@ export class AuthService {
     return user;
   }
 
+  async generateAccountActivationToken(userId: number): Promise<string> {
+    const token = await this.generateBearerToken(
+      userId,
+      process.env.JWT_ACCOUNT_ACTIVATION_SECRET,
+      +process.env.ACCOUNT_ACTIVATION_TIME,
+    );
+    await this.userRepository.saveAccountActivationToken(userId, token);
+
+    const timeInMilliseconds = +process.env.ACCOUNT_ACTIVATION_TIME * 1000;
+    const timeoutName = this.accountActivationTimeouts.getName(userId);
+
+    this.accountActivationTimeouts.addTimeout(
+      userId,
+      timeInMilliseconds,
+      timeoutName,
+    );
+
+    return token;
+  }
+
+  async verifyAccountActivationToken(
+    jwtToken: string,
+  ): Promise<{ id: number }> {
+    try {
+      return this.jwtService.verifyAsync(jwtToken, {
+        secret: process.env.JWT_ACCOUNT_ACTIVATION_SECRET,
+      });
+    } catch {
+      throw new ForbiddenException('Token expired');
+    }
+  }
+
+  async activateAccount(userId: number): Promise<User> {
+    const timeoutName = this.accountActivationTimeouts.getName(userId);
+    try {
+      this.accountActivationTimeouts.deleteTimeout(timeoutName);
+    } catch {
+      throw new NotFoundException(
+        `Account deletion timeout has been already deleted, therefore your accound is already activated.`,
```

Comment:
As a user I have no idea what it means 😅 I even have trouble understanding it as a developer 😅 

Why "Account deletion timeout" deleted => account activated? What if account was not activated in time and it has been deleted? Or does it not work this way?

Reply:
also, `therefore your accound` -> `therefore your account` - always double check the PRs! :)